rflx/expression.py

```diff
@@ -1546,7 +1550,8 @@ def substituted(
         self, func: Callable[["Expr"], "Expr"] = None, mapping: Mapping["Name", "Expr"] = None
     ) -> "Expr":
         return self.__class__(
-            self.lower.substituted(mapping=mapping), self.upper.substituted(mapping=mapping)
+            self.lower.substituted(func=func, mapping=mapping),
+            self.upper.substituted(func=func, mapping=mapping),
         )
```

Comment:
The behavior of this function is different to other expressions. It should be possible to substitute the whole expression. Please have a look at `Slice.substituted` for an example.

Reply:
Fixed by https://github.com/Componolit/RecordFlux/pull/381/commits/0eb7b428aaeed454b876b131aa7108734b8bc643.